internal/controllers/topology/cluster/desired_state.go

```diff
@@ -212,6 +212,13 @@ func computeControlPlane(_ context.Context, s *scope.Scope, infrastructureMachin
 		}
 	}
 
+	// If it is required to manage the number of replicas for the control plane, set the corresponding field.
+	if s.Blueprint.Topology.ControlPlane.NodeDrainTimeout != nil {
+		if err := contract.ControlPlane().NodeDrainTimeout().Set(controlPlane, *s.Blueprint.Topology.ControlPlane.NodeDrainTimeout); err != nil {
```

Comment:
```suggestion
		if err := contract.ControlPlane().MachineTemplate().NodeDrainTimeout().Set(controlPlane, *s.Blueprint.Topology.ControlPlane.NodeDrainTimeout); err != nil {
```
see: https://cluster-api.sigs.k8s.io/developer/architecture/controllers/control-plane.html#required-spec-fields-for-implementations-using-machines